app/models/merchant.rb

```diff
@@ -30,4 +30,12 @@ def self.most_items(num_of_merchants)
       order("most_items DESC").
       limit(num_of_merchants)
   end
+
+  def customers_with_pending_invoices
+    invoices.
+      joins(:transactions).
+      where(transactions: {result: "failed"}).
+      distinct.map &:customer
```

Comment:
Let's try to change that to an active record query, once we get all the business intelligence endpoints to work